crates/rspack_loader_swc/src/compiler.rs

```diff
@@ -346,16 +350,37 @@ impl SwcCompiler {
     }
   }
 
-  pub fn transform(&self, config: BuiltInput<impl Pass>) -> Result<Program, Error> {
+  pub fn transform(&self, config: BuiltInput<impl Pass>) -> Result<Program, miette::Report> {
     let program = config.program;
     let mut pass = config.pass;
 
     let program = self.run(|| {
       helpers::HELPERS.set(&self.helpers, || {
-        try_with_handler(self.cm.clone(), Default::default(), |handler| {
-          HANDLER.set(handler, || Ok(program.apply(&mut pass)))
-        })
-      })
+      let result = try_with_handler(self.cm.clone(), Default::default(), |_handler| {
+        let result = program.apply(&mut pass);
+        Ok(result)
+      });
+      match result {
+        Ok(v) => Ok(v),
+        Err(err) => {
+        let error_msg = match err.downcast_ref::<String>(){
+          Some(msg) => {
+            msg
+          },
+          None => "unknown error"
+        };
+        // FIXME: with_help has bugs, use with_help when diagnostic print is fixed
+        let help_msg =  indoc!{"
+          The version of the SWC Wasm plugin you're using might not be compatible with `builtin:swc-loader`.
+          The `swc_core` version of the current `rspack_core` is {swc_core_version}. 
+          Please check the `swc_core` version of SWC Wasm plugin to make sure these versions are within the compatible range.
+          Versions of `swc_core` are more likely to be located in the `Cargo.toml` file in the root directory of these plugin repositories.
```

Comment:
Maybe we can remove this line. It is hard for normal JS developers to understand, and we have mentioned this in the FAQ doc.

```suggestion
```